microsoft-edge/extensions-chromium/developer-guide/native-messaging.md

```diff
@@ -22,6 +22,25 @@ To acquire the extension and native app host, there are two different distributi
 To create your extension to send and receive messages with native app hosts, do the following steps.
 
 
+<!-- ====================================================================== -->
+## Writing the chrome_native_messaging_host.exe app
+
+Writing the chrome_native_messaging_host.exe app - this section provides code examples in multiple programming languages.
+
+How you actually write the chrome_native_messaging_host.exe application.
+
+stdin and stdout are used to communicate with the host.
+
+How the information sent from the extension is encoded.
+
+Perhaps:
+The first four bytes is a number that specifies packet length, and the rest is the packet data encoded as JSON.
+
+How to encode the information that is sent back to the extension: tbd
+
+How the chrome_native_messaging_host.exe application knows when to exit: tbd
+
+
 <!-- ====================================================================== -->
 ## Step 1 - Add permissions to the extension manifest
 
```

Comment:
todo: update the "here" sentence, 404
```
The original page is found [here](https://developer.chrome.com/extensions/nativeMessaging).
```
https://developer.chrome.com/extensions/nativeMessaging

one time, searching web for that url caused ending up at:
https://developer.chrome.com/docs/extensions/develop/concepts/native-messaging